examples/src/main/scala/org/apache/spark/examples/sql/streaming/StructuredKafkaWordCount.scala

```diff
@@ -46,11 +51,13 @@ object StructuredKafkaWordCount {
   def main(args: Array[String]): Unit = {
     if (args.length < 3) {
       System.err.println("Usage: StructuredKafkaWordCount <bootstrap-servers> " +
-        "<subscribe-type> <topics>")
+        "<subscribe-type> <topics> [<checkpoint-location>]")
       System.exit(1)
     }
 
-    val Array(bootstrapServers, subscribeType, topics) = args
+    val Array(bootstrapServers, subscribeType, topics, _*) = args
+    val checkpointLocation =
+      if (args.length > 3) args(3) else "/tmp/temporary-" + UUID.randomUUID.toString
```

Comment:
why bother supplying a default?  will this be any better than spark's internal default?

Reply:
This is what the internal default would be if java.io.tmpdir is "/tmp", but in case of YARN cluster mode, java.io.tmpdir is something else (the underlying problem). Supplying this default here is just to ease the user experience. They would get the same result running in YARN cluster mode or client mode, without supplying an explicit checkpoint location.

Reply:
talked to wing yew about this offline -- so the other part of the story is that in yarn it will by default by a dir on hdfs, and in general /tmp is writeable on hdfs.  And somewhat surprisingly, if you run `fs.create()` on a path where the parent dir doesn't exist, hdfs will create the parents.